noobaa/noobaa-operator
# Support AWS STS in Noobaa

### Explain the changes
1. Added spec field for backing stores crd to include Role ARN for AWS STS
2. Added spec field for noobaa crd to include Role ARN for AWS STS
3. Added Projected Service account token mount to noobaa-operator and nooba-core pods
4. Added function to create AWS STS backing store
5. Added noobaa install cli option `--aws-sts-role-arn`
6. Added check for AWS STS cluster
7. Added Admission Webhook validation for BackingStore
8. Added Validation check for NamespaceStore to not include AWS STS ARN
Signed-off-by: Kaustav Majumder <kmajumde@redhat.com>

### Testing Instructions:
1. Create an STS based Kubernetes or Openshift cluster [see more..](https://docs.openshift.com/container-platform/4.7/authentication/managing_cloud_provider_credentials/cco-mode-sts.html#sts-mode-installing-manual-config_cco-mode-sts)
2. Create Role for S3 full access and assign to Identity Provider which was created during cluster install
3. Use noobaa cli to pass in the ARN for the role created previously 
`nb install --aws-sts-role-arn=arn:aws:iam::261532230807:role/kmajumder_noobaa_sts`
4. Check if the default backing store reaches ready state.
